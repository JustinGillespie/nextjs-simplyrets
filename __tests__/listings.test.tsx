/**
 * @jest-environment jsdom
 */

import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import Page from "../pages/listings/index";

describe("Listings page", () => {
  it("renders the global header", () => {
    render(<Page />);
    const heading = screen.getByTestId("global-header");
    expect(heading).toBeInTheDocument();
  });

  it("renders the listing container component", () => {
    render(<Page />);
    const listingsContainer = screen.getByTestId("listings");
    expect(listingsContainer).toBeInTheDocument();
  });

  it("renders listing page unchanged", () => {
    const { container } = render(<Page />);
    expect(container).toMatchSnapshot();
  });
});
