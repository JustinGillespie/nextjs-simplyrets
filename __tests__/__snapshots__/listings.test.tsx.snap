// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Listings page renders listing page unchanged 1`] = `
<div>
  <div
    class="page"
  >
    <header
      class="header"
      data-testid="global-header"
      role="header"
    >
      <h1>
        Property Listings
      </h1>
    </header>
    <div
      data-testid="listings"
      role="propertylistings"
    >
      <div
        class="flex"
      >
        <section
          class="column fill"
          role="propertylistingsmap"
        />
        <section
          class="column fill hidden"
          role="propertyresultslist"
        >
          <div
            class="listings"
          >
            Loading...
          </div>
        </section>
      </div>
      <button
        class="button"
      >
        List View
      </button>
    </div>
  </div>
</div>
`;
